src/nginxconfig/templates/app.vue

```diff
@@ -220,6 +220,20 @@ THE SOFTWARE.
             // Fallback to the window hash if no search query params, from the Angular version of nginxconfig
             // The config file watcher will handle setting the app as ready
             const query = window.location.search || window.location.hash.slice(1);
+            // Adaptive system language,The system language is not read after the language is set
+            if (!query.includes('global.app.lang') && typeof navigator !== undefined){
+                if (typeof navigator.language === 'string'){
+                    let lang = navigator.language.match(/(\w+)/g).map(function(value){
+                        return value.toUpperCase();
+                    });
+                    lang[0] = lang[0].toLowerCase();
+                    lang = lang.join('');
+                    if (lang in i18nPacks){
+                        this.$data.global.app.lang.value = lang;
+                        this.$data.global.app.lang.computed = lang;
+                    }
+                }
```

Comment:
```suggestion
                const rawLangs = new Set();
                
                // Get the user language
                if (Array.isArray(window.navigator.languages))
                    window.navigator.languages.forEach(lang => langs.add(lang));
                if (typeof window.navigator.language === 'string')
                    langs.add(window.navigator.language);
                if (Intl && 'DateTimeFormat' in Intl)
                    langs.add(Intl.DateTimeFormat().resolvedOptions().locale);
                
                const toPack = lang => lang.split('-')[0].toLowerCase() + lang.split('-').slice(1).map(x => x.toUpperCase()).join('-');
                const i18nPackLangs = i18nPacks.reduce((acc, data) => {
                    const lang = data.match(/^[a-z]+/)[0];
                    if (!(lang in acc)) acc[lang] = data;
                    return acc;
                }, {});
                
                // Try to get an exact region match, or fallback to language match
                const exactMatch = [...rawLangs.values()].find(x => i18nPacks.includes(toPack(x)));
                const langMatch = [...rawLangs.values()].find(x => i18nPackLangs.includes(x.split('-')[0].toLowerCase()));
                
                if (exactMatch || langMatch) {
                    this.$data.global.app.lang.value = exactMatch || i18nPackLangs[langMatch];
                    this.$data.global.app.lang.computed = exactMatch || i18nPackLangs[langMatch];
                }
```
I haven't tested this, but I think this will help with giving the best chances of making a positive region/language match.

Reply:
Cool!!!You're awesome